src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/DisableRuntimeMarshallingTests.cs

```diff
@@ -0,0 +1,1404 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.InteropServices.DisableRuntimeMarshallingAnalyzer,
+    Microsoft.NetCore.Analyzers.InteropServices.CSharpDisableRuntimeMarshallingFixer>;
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.InteropServices.DisableRuntimeMarshallingAnalyzer,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Microsoft.CodeAnalysis.NetAnalyzers.UnitTests.Microsoft.NetCore.Analyzers.InteropServices
+{
+    public class DisableRuntimeMarshallingTests
+    {
+        [Fact]
+        public async Task CS_PInvokeWithSetLastError_Emits_Diagnostic()
+        {
+            string source = @"
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+
+[assembly:DisableRuntimeMarshalling]
+
+class Foo
+{
+    [DllImport(""Foo"", SetLastError = true)]
+    public static extern void {|CA1420:Bar|}();
+}
+";
+            await VerifyCSAnalyzerAsync(source);
+        }
+
+        [Fact]
+        public async Task VB_PInvokeWithSetLastError_Emits_Diagnostic()
+        {
+            string source = @"
+Imports System.Runtime.CompilerServices
+Imports System.Runtime.InteropServices
+<assembly:DisableRuntimeMarshalling>
+Public Class Foo
+    <DllImport(""foo"", SetLastError:=True)>
+    Public Shared Sub {|CA1420:Bar|}()
+    End Sub
+End Class
+";
+            await VerifyVBAnalyzerAsync(source);
+        }
+
+        [Fact]
+        public async Task CS_PInvokeWithLCIDConversion_Emits_Diagnostic()
+        {
+            string source = @"
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+
+[assembly:DisableRuntimeMarshalling]
+
+class Foo
+{
+    [DllImport(""Foo"")]
+    [LCIDConversion(0)]
+    public static extern void {|CA1420:Bar|}();
+}
```

Comment:
Consider testing a local function and validate if the behavior for it is correct.